Add support for defining Vault address and token through environment variables in Consul Connect CA provider (#17981)
hashicorp/consul: website/content/docs/connect/ca/vault.mdx

  This token must have [proper privileges](#vault-acl-policies) for the PKI
  paths configured. In Consul 1.8.5 and later, if the token has the [renewable](/vault/api-docs/auth/token#renewable)
  flag set, Consul will attempt to renew its lease periodically after half the

>   duration has expired. You can also provide this token through an environment variable `VAULT_TOKEN`. Token has to be provided either through the configuration file or through the environment variable. Both cannot be used at the same time.

Reviewer: ```suggestion
  duration has expired. You can also provide the token through the `VAULT_TOKEN` environment variable. The token must be provided either through the configuration file or through the environment variable, but you cannot provide the token using both methods to at the same time.
```

Author: Thanks!
